apis/addons/v1alpha1/addons_types.go

```diff
@@ -6,13 +6,18 @@ import (
 
 // AddonSpec defines the desired state of Addon.
 type AddonSpec struct {
+	// Human readable name for this addon
```

Comment:
> We can additionally add `json:"displayName,omitempty"` AND // `+kubebuilder:validation:Required=true` to prevent marshaling empty strings on the wire.

What would that give us? An empty string would not pass parsing validation anyways when MinLength is 1, right?

Reply:
`,omitempty`
Would just remove the key being printed when outputting json, but also marks the field as optional.
So we have to make it required again by adding the kubebuilder `Required` annotation, otherwise the validation would not run when the value is not present.

As mentioned that's just something additional that came to my mind and in no way required.